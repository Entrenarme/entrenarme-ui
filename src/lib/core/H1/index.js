// @flow
import * as React from 'react';
import styled from 'styled-components';

import * as fonts from '../../helpers/fonts';

const StyledH1 = styled.h1`
  text-transform: uppercase;
  font-size: 32px;
  font-weight: 700;
  font-family: ${fonts.mainFont};
`;

type Props = {
  children: React.Node,
};

const H1 = ({ children, ...rest }: Props) => (
  <StyledH1 {...rest}>{children}</StyledH1>
);

H1.defaultProps = {};

export default H1;
