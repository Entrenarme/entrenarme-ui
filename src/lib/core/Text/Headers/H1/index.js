// @flow
import * as React from 'react';
import styled from 'styled-components';

import * as fonts from '../../../../helpers/fonts';
import colors from '../../../../helpers/colors';

const normalDesktop = 2;

const normalMobile = 1.5;

const sizeModifier = 0.25;

const textSizeDesktopTransform = props => {
  if (props.l) {
    return normalDesktop + sizeModifier;
  } else if (props.s) {
    return normalDesktop - sizeModifier;
  } else {
    return normalDesktop;
  }
};

const textSizeMobileTransform = props => {
  if (props.l) {
    return normalMobile + sizeModifier;
  } else if (props.s) {
    return normalMobile - sizeModifier;
  } else {
    return normalMobile;
  }
};

const StyledH1 = styled.h1`
  text-transform: uppercase;
  font-size: ${props => `${textSizeDesktopTransform(props)}rem`};
  font-weight: 700;
  font-family: ${fonts.mainFont};
  color: ${props => (props.color ? props.color : colors.primary.default)};

  /* Smartphones (portrait) ----------- */
  @media only screen and (max-width: 320px) {
    font-size: ${props => `${textSizeMobileTransform(props)}rem`};
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (min-width: 320px) and (max-width: 767px) {
    font-size: ${props => `${textSizeMobileTransform(props)}rem`};
  }

  /* iPads (portrait and landscape) ----------- */
  @media only screen and (min-width: 768px) and (max-width: 1223px) {
    /* Styles */
  }

  /* Desktops and laptops ----------- */
  @media only screen and (min-width: 1224px) {
    /* Styles */
  }

  /* Large screens ----------- */
  @media only screen and (min-width: 1824px) {
    /* Styles */
  }

  /* ----------- Retina Screens ----------- */
  @media screen and (min-device-width: 1200px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
    /* Styles */
  }
`;

type Props = {
  children: React.Node,
  color?: string,
};

const H1 = ({ children, ...rest }: Props) => (
  <StyledH1 {...rest}>{children}</StyledH1>
);

H1.defaultProps = {};

export default H1;
