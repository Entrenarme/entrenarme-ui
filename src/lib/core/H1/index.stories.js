// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import H1 from './index';

storiesOf('H1', module).add(
  'default',
  withInfo('Heading 1')(() => <H1>Heading 1</H1>),
);
