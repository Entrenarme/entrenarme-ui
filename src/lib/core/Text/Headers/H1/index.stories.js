// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import H1 from './index';

const ElementDecoration = storyFn => (
  <div style={{ paddingLeft: '15px' }}>{storyFn()}</div>
);

storiesOf('Text/Headers/H1', module)
  .addDecorator(ElementDecoration)
  .add('default', withInfo('Heading 1')(() => <H1>Heading 1</H1>));

storiesOf('Text/Headers/H1', module)
  .addDecorator(ElementDecoration)
  .add('small', withInfo('Heading 1')(() => <H1 s>Heading 1</H1>));

storiesOf('Text/Headers/H1', module)
  .addDecorator(ElementDecoration)
  .add('large', withInfo('Heading 1')(() => <H1 l>Heading 1</H1>));

storiesOf('Text/Headers/H1', module)
  .addDecorator(ElementDecoration)
  .add(
    'with color prop',
    withInfo('Heading 1')(() => <H1 color="red">Heading 1</H1>),
  );
