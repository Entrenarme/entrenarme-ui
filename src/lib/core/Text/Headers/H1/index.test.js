import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import H1 from './index';

test('renders "Heading" with no props', () => {
  const { container } = render(<H1>Heading</H1>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with large size', () => {
  const { container } = render(<H1 size="large">Heading</H1>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with small size', () => {
  const { container } = render(<H1 size="small">Heading</H1>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with color prop', () => {
  const { container } = render(<H1 color="warning">Heading</H1>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with text align', () => {
  const { container } = render(<H1 textAlign="center">Heading</H1>);
  expect(container.firstChild).toMatchSnapshot();
});
