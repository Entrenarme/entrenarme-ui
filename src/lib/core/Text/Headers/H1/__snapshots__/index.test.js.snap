// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Heading" with color prop 1`] = `
<h1
  class="sc-bdVaJa gIEXBz"
>
  Heading
</h1>
`;

exports[`renders "Heading" with large size 1`] = `
<h1
  class="sc-bdVaJa bluQvN"
>
  Heading
</h1>
`;

exports[`renders "Heading" with no props 1`] = `
<h1
  class="sc-bdVaJa cmdThp"
>
  Heading
</h1>
`;

exports[`renders "Heading" with small size 1`] = `
<h1
  class="sc-bdVaJa TtggM"
>
  Heading
</h1>
`;

exports[`renders "Heading" with text align 1`] = `
<h1
  class="sc-bdVaJa ggFWVx"
>
  Heading
</h1>
`;
